google/resource_pubsub_topic.go

```diff
@@ -76,3 +80,20 @@ func resourcePubsubTopicDelete(d *schema.ResourceData, meta interface{}) error {
 
 	return nil
 }
+
+func resourcePubsubTopicStateImporter(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return nil, err
+	}
+
+	name := d.Id()
+	id := fmt.Sprintf("projects/%s/topics/%s", project, name)
```

Comment:
No need to overwrite the id here. The id should contain only the `name`. See the id format in the create method. 

This is because d.Id() is being use in the read and delete method and the value is assumed to be name.

After moving the d.Set("name", ...) to the read method like suggested below, you can remove this method entirely and only use the passthrough handy function:

```go
Importer: &schema.ResourceImporter{
			State: schema.ImportStatePassthrough,
		},
```

Reply:
Moving d.Set("name", ...) to the read method makes sense, but if we move to using the passthrough import function the user would have to specify a much longer id during import - "projects/my-project/topics/my-topic", rather than just "my-topic", where the import method then interpolates the project name from the manifests.
I would personally prefer to just specify the name during import, but if the more experienced devs here prefer the longer form, I will change the PR to reflect this.
For now I have updated the PR to have the read method set the name instead of the import method doing it.
Let me know which way you want to go with this :)

Reply:
I am working on reusable id parser that will support both the self_link or just the name  (inferring the project from the config) for a resource. This will be basically a generalization of what I did here: https://github.com/terraform-providers/terraform-provider-google/pull/378/files#diff-c969a5ee914c2e0cc17cc445f4b5ee22R169. No need for you to worry about this now. I will update the import method for all resources in one swoop. For now, just support the import by name only case.

The id format is currently only the `name`. It means that if you use a pass-through, then the import will require ONLY the name from the user. The current id format we use in Terraform for pubsub_topic is NOT "projects/%s/topics/%s". "projects/%s/topics/%s" is the relative part of the self_link of the resource. self_link != id (although they could be but this is not what is currently used for most resource).

Basically, you can delete your `resourcePubsubTopicStateImporter` method and use the pass-through and import will work by name:
```go
Importer: &schema.ResourceImporter{
    State: schema.ImportStatePassthrough,
},
```